pkg/action/archive.go

```diff
@@ -58,8 +60,13 @@ func extractTar(ctx context.Context, d string, f string) error {
 			return fmt.Errorf("failed to create xz reader: %w", err)
 		}
 		tr = tar.NewReader(xzStream)
+	case strings.Contains(filename, ".bz2") || strings.Contains(filename, ".bzip2"):
+		if err != nil {
+			return fmt.Errorf("failed to seek to start: %w", err)
+		}
+		br := bzip2.NewReader(tf)
```

Comment:
It's weird that `bzip2.NewReader` only returns an `io.Reader` instead of `(io.Reader, err)`. It would be nice to do some error checking like we do with `gzip.NewReader`. 😞